### ISSUE TITLE
[Messenger] Can't repeat #[AsMessageHandler] to replicate previous functionality of deprecated MessageSubscriberInterface

### ISSUE THREAD

**Issue by adrolter** [symfony/symfony#50267]
### Symfony version(s) affected

6.2.x

### Description

Using the `MessageSubscriberInterface` it was possible to declare multiple handlers pointing to different methods of a class, without the implementation of those methods being directly defined in that class — e.g.:

```php
class FooHandler 
    extends AbstractHandler // <- contains implementations
    implements MessageSubscriberInterface
{
    public function getHandledMessages(): iterable
    {
        yield SomeMessage::class => ['method' => 'someMethodFromAbstractParent'];
        yield SomeOtherMessage::class => ['method' => 'someOtherMethodFromAbstractParent'];
    }
}
```

Unless I'm missing something obvious this is no longer possible, as `#[AsMessageHandler]` would have to be repeatable on the class itself to replicate this functionality, but the attribute lacks  the `IS_REPEATABLE` flag.

### Possible Solution

Add the `\Attribute::IS_REPEATABLE` flag to the existing flags on the attribute, supporting the following syntax:

```php
#[AsMessageHandler(
    handles: SomeMessage::class,
    method: 'someMethodFromAbstractParent',
    priority: 10,
)]
#[AsMessageHandler(
    handles: SomeOtherMessage::class,
    method: 'someOtherMethodFromAbstractParent',
    priority: 10,
)]
class FooHandler 
    extends AbstractHandler // <- contains implementations
{
}
```

This seems to resolve the issue, and I'm happy to submit a PR if it would be accepted.

**Comment by chalasr** [comment]
Can you please tell more about your use case? Handling different messages from the same handler is mixing concerns to me.

**Comment by derrabus** [comment]
A possible use-case: In one project I've worked on, different message classes were being used to indicate different priorities, e.g. I had a `MyMessage` and a `MyHighPriorityMessage`. They eventually were both routed to the same handler, but the queue configuration was different.

To me, it indeed looks like an accident that the attribute is not repeatable.

**Comment by stof** [comment]
I agree here. There is no reason not to make it repeatable as the component handles properly the case of having multiple messages handled by the same service.

**Comment by chalasr** [comment]
Indeed 🙌

**Comment by stof** [comment]
@adrianguenter do you want to open a PR implementing this ?

**Comment by adrolter** [comment]
@stof Will do

**Comment by adrolter** [comment]
@chalasr An argument could certainly be made that my approach to this specific problem doesn't stand up 100% to SOLID by the most rigorous of textbook standards (then again, what software does? ), but...this functionality previously existed and I found no documentation of its removal, so I made my design decisions under the mistaken assumption that the attribute would support it. I'd happily have refactored my code if this change wasn't desired, though!
